org.sonarlint.eclipse.jdt/src/org/sonarlint/eclipse/jdt/internal/JdtUtils.java

```diff
@@ -60,11 +63,21 @@ static boolean hasJavaNature(IProject project) {
   /**
    * SLE-34 Remove Java files that are not compiled.This should automatically exclude files that are excluded / unparseable. 
    */
-  public static boolean isValidJavaFile(IFile file) {
-    boolean hasJavaNature = hasJavaNature(file.getProject());
-    IJavaProject javaProject = JavaCore.create(file.getProject());
+  public static boolean shouldExclude(IFile file) {
     IJavaElement javaElt = JavaCore.create(file);
-    return javaElt == null || (hasJavaNature && isStructureKnown(javaElt) && javaProject.isOnClasspath(javaElt));
+    if (javaElt == null) {
+      // Not a Java file, don't exclude it
+      return false;
+    }
+    if (!javaElt.exists()) {
+      // SLE-218 Visual Cobol with JVM Development make JDT think .cbl files are Java files. But still we want to analyze them, so only
+      // exclude
+      // files having the original java source content type.
```

Comment:
Cosmetic, but this comment would be easier to read if layed out like this:

    // SLE-218 Visual Cobol with JVM Development make JDT think .cbl files are Java files.
    // But still we want to analyze them, so only exclude files having the original java source content type.